### PR TITLE
Update anthropic models

### PR DESCRIPTION
**Pull request by ravi03071991**
# Description

PR to update latest anthropic models.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by logan-markewich — COMMENTED** [review]
Can you bump the version in the pyproject.toml file of this package? Then it will automatically publish when merging

File: llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py [hunk 1]
```diff
@@ -143,15 +144,17 @@ def _get_all_kwargs(self, **kwargs: Any) -> Dict[str, Any]:
...
-        prompt = messages_to_anthropic_prompt(messages)
+        anthropic_messages, system_prompt = messages_to_anthropic_messages(messages)
         all_kwargs = self._get_all_kwargs(**kwargs)
 
-        response = self._client.completions.create(
-            prompt=prompt, stream=False, **all_kwargs
+        response = self._client.messages.create(
+            messages=anthropic_messages,
+            stream=False,
+            system_prompt=system_prompt**all_kwargs,
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/base.py:153]: missing a comma here for all_kwargs

File: llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/utils.py [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-
-
 CLAUDE_MODELS: Dict[str, int] = {
     "claude-instant-1": 100000,
     "claude-instant-1.2": 100000,
     "claude-2": 100000,
     "claude-2.0": 100000,
     "claude-2.1": 200000,
+    "claude-3-opus-20240229": 200000,
+    "claude-3-sonnet-20240229": 200000,
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/utils.py:12]: is there also a `haiku` model?

Comment (ravi03071991) [reply llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/utils.py:12]: haiku model not yet released.

**Review by logan-markewich — APPROVED** [review]

**Review by jerryjliu — COMMENTED** [review]
can we update notebook example to at least use sonnet? 

https://docs.llamaindex.ai/en/stable/examples/llm/anthropic.html

(can be in a later PR)

**Comment by ravi03071991** [comment]
@jerryjliu making an update to the notebook

**Pull request merged**